### ISSUE TITLE
[Proposal] Web Support for fl_chart

### ISSUE THREAD

**Issue by AbhishekDoshi26** [imaNNeoFighT/fl_chart#780]
The package is working perfectly for web, but on pub.dev web platform is not showing. I forked the repo and ran `flutter create .` but it's not creating web folder. So maybe, you can try out the same and enable flutter web support and then push and publish on pub.dev again thereby providing web support!

**Comment by imaNNeoFighT** [comment]
Great! I didn't know that. I will work on it. Thanks for letting me know.

**Comment by AbhishekDoshi26** [comment]
@imaNNeoFighT Np! But do you know the reason why `flutter create .` didn't create the web folder?
Are there any explicit rules added in the project which didn't allow the analyzer to create?
Not just me, one of my friends also tried and failed.
My flutter version: 2.2.3
His Flutter Version: 2.5.1

**Comment by imaNNeoFighT** [comment]
I think we should write this in our pubspec.yaml.

```
flutter:
  plugin:
    platforms:
      web:
```

**Comment by AbhishekDoshi26** [comment]
Okk let me try this and create PR for the same

**Comment by imaNNeoFighT** [comment]
Oh got it.

```
Consider supporting multiple platforms:

Package not compatible with runtime flutter-web on Web
Because:

package:fl_chart/fl_chart.dart that imports:
package:fl_chart/src/chart/scatter_chart/scatter_chart.dart that imports:
package:fl_chart/src/chart/scatter_chart/scatter_chart_renderer.dart that imports:
package:fl_chart/src/chart/scatter_chart/scatter_chart_painter.dart that imports:
package:fl_chart/src/chart/base/base_chart/base_chart_painter.dart that imports:
package:fl_chart/src/chart/base/base_chart/base_chart_data.dart that imports:
package:fl_chart/src/chart/base/base_chart/fl_touch_event.dart that imports:
dart:io
```

**Comment by AbhishekDoshi26** [comment]
Yeah, dart:io isn't supported for Web. Maybe you can use https://pub.dev/packages/universal_io as replacement for dart:io

**Comment by imaNNeoFighT** [comment]
Yes, we need to check why we needed dart:io there

**Comment by AbhishekDoshi26** [comment]
Did you use Transform or something similar to that?

**Comment by AbhishekDoshi26** [comment]
I will replace the package and see if there's any side effects

**Comment by imaNNeoFighT** [comment]
No, I just use Platforms.isLinux, ...

Thanks!

**Comment by imaNNeoFighT** [comment]
BTW It's better to not use a package to solve this (If we can solve it without needing to importing a package)

**Comment by AbhishekDoshi26** [comment]
Agreed.

**Comment by AbhishekDoshi26** [comment]
@imaNNeoFighT So just to check, I removed `import 'dart:io';` and to changed the ` final isDesktopOrWeb = kIsWeb;` but it is still not creating the web folder. So it's not an issue of dart:io

**Comment by AbhishekDoshi26** [comment]
I also tried by adding the following but still no luck

> I think we should write this in our pubspec.yaml.
> 
> ```
> flutter:
>   plugin:
>     platforms:
>       web:
> ```

**Comment by imaNNeoFighT** [comment]
@AbhishekDoshi26 The problem is not the existence of the web directory.
(Actually, there is no other `linux`, `android`, or any other directories) 
The directory is not an indicator of supporting it. If we can remove dart:io, pub.dev will indicate our library to support web automatically.

**Comment by AbhishekDoshi26** [comment]
@leidig54 Actually the package works perfectly well on web. The only thing is, due to some reason, it is not showing web support on pub.dev

So you can use it for web without any issues.
